src/main/java/net/wurstclient/commands/DistCmd.java

```diff
@@ -0,0 +1,94 @@
+package net.wurstclient.commands;
+
+import net.minecraft.util.math.BlockPos;
+import net.wurstclient.command.CmdException;
+import net.wurstclient.util.ChatUtils;
+import net.wurstclient.command.CmdError;
+import net.wurstclient.command.CmdException;
+import net.wurstclient.command.CmdSyntaxError;
+import net.wurstclient.command.Command;
+import java.lang.Math;
+
+
+public class DistCmd extends Command {
+
+
+    public DistCmd() {
+        super("dist", "Gets Distance between 2 Cordinates\nYou can convert Overworld Cords To Nether cords With cvt:<x> <z>\n ", ".dist <x> <z>:<x> <z>", ".dist cvt:<x> <z>");
+    }
+
+    @Override
+    public void call(String[] args) throws CmdException {
+
+        if(args.length < 1)
+            throw new CmdError("Enter Command Properly");
+
+        String Allcords = String.join(" ", args);
+        if(!Allcords.contains(":"))
+            throw new CmdError("Separate Cordinates With : ");
+
+        if (Allcords.startsWith("cvt")){
+
+            String[] arrOfSTR = Allcords.split(":", 0);
+            String cordsWithSpace = arrOfSTR[1];
+            String[] Cordsarray = cordsWithSpace.split(" ",0);
+            if (Cordsarray.length < 2)
+                throw new CmdError("Provide Cordinates Properly");
+            int cvtX;
+            int cvtZ;
+            cvtX = Integer.parseInt(Cordsarray[0]);
+            cvtZ = Integer.parseInt(Cordsarray[1]);
+
+            int netherX =  cvtX / 8;
+            int netherZ = cvtZ / 8;
+            ChatUtils.message("Nether cords For "+ cvtX + " "+ cvtZ + " is " + netherX + " " + netherZ);
+        }else {
+            String[] arrOfStr = Allcords.split(":", 0);
+            String FromCord;
+            String ToCord;
+            FromCord = arrOfStr[0];
+            ToCord = arrOfStr[1];
+            String[] FromCordStrip = FromCord.split(" ",0);
+            String[] ToCordStrip = ToCord.split(" ", 0);
+            if (FromCordStrip.length < 2 || ToCordStrip.length < 2)
+                throw new CmdError("Provide Cordinates Properly");

```

Comment:
We can simplify this as we will always know the position of the args. We can basically just check for the proper length here, but also don't forget when you use the tilde for current pos, your length conditional fails.

Reply:
i have tried my best to fix all the issues and optimize it as you suggested and thanks for the Motivation it really helped me😃

Reply:
At first glance it looks so much better. 😄 
I'll take a more in-depth look in a little while. I knew you had it in you 😎